src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.cs

```diff
@@ -4520,11 +4510,102 @@ private ITypeSymbol GetXBindPropertyPathType(string propertyPath, INamedTypeSymb
 				}
 				else
 				{
-					throw new InvalidOperationException($"Unable to find member [{part}] on type [{currentType}]");
+					// We can't find the type. It could be something that is source-generated, or it could be a user error.
+					// For source-generated members, it's not possible to reliably get this information due to https://github.com/dotnet/roslyn/issues/57239
+					// However, we do a best effort to handle the common scenario, which is code generated by CommunityToolkit.Mvvm.
+
+					// Case 1: ObservableProperty attribute.
+					// Relevant code: https://github.com/CommunityToolkit/dotnet/blob/b341ef91fe66101444d6b811bc5dff32de029d3c/src/CommunityToolkit.Mvvm.SourceGenerators/ComponentModel/ObservablePropertyGenerator.Execute.cs#L1231-L1244
+					// The user defines a field named '_abc', 'abc', 'm_abc', '_Abc', or 'm_Abc'.
+					// The generated property will be named 'Abc', which is what the user will be referencing in x:Bind.
+					ITypeSymbol? mvvmType = TryGetMvvmObservablePropertyType(currentType, part);
+
+					// Case 2: RelayCommand attribute
+					// Relevant code: https://github.com/CommunityToolkit/dotnet/blob/b341ef91fe66101444d6b811bc5dff32de029d3c/src/CommunityToolkit.Mvvm.SourceGenerators/Input/RelayCommandGenerator.Execute.cs#L455-L496
+					// and also: https://github.com/CommunityToolkit/dotnet/blob/b341ef91fe66101444d6b811bc5dff32de029d3c/src/CommunityToolkit.Mvvm.SourceGenerators/Input/RelayCommandGenerator.Execute.cs#L522-L614
+					// The user defines a method named 'OnAbcAsync', 'AbcAsync', 'OnAbc', 'Abc'.
+					// The generated command property will be named 'AbcCommand'
+					mvvmType ??= TryGetMvvmRelayCommandType(currentType, part);
```

Comment:
I wonder if we should be having some some of addin system to work around this, so that we can allow for third party libraries to hint our tooling somehow.

Reply:
It will be quite hard. The only way I can think of to change generator behavior dynamically is by reading an additional file. It doesn't seem viable to have any sort of robust format that can encode this info that can work for all libraries.

Do you have something in mind?

Reply:
If there are packages that can provide single DLLs, discoverable by additional itemgroup, that we know how to load (maybe through an implicit contract to avoid having another package, we could run that code by giving it access to the compilation object. It's a bit weak, but it gives flexibility for other generators or apps themselves (to adjust for libraries that may not provide such add-ins).

Reply:
I agree that we should not be doing this, yet this is going to be an increasingly problematic issue. We may be doing the "add-ins" ourselves for libraries, that's not too much of a problem.

Reply:
> "I'm also not sure if CommunityToolkit.Mvvm can/will want to create such a dll"

I mean the analyzer is just packed as a .dll into the MVVM Toolkit NuGet package, so nothing's preventing anyone from getting it and manually loading it somewhere else. We wouldn't invest time into making other special-purpose versions of our analyzers though, if that's what you meant.

Reply:
@jeromelaban I'm not following.

Couple of points:

- The fact that we can't know generated members is actually "by-design" but given the popularity of CommunityToolkit.Mvvm and the fact that the generated properties are commonly used in `x:Bind`, we need that support.
- I believe source generators weren't design (at the compiler side) with this level of extensibility in mind


I'm assuming that you are suggesting that `CommunityToolkit.Mvvm` will provide us with the information we need, and we will be completely unaware of the implementation details.

Here, there are two levels:

- Something extensible by anyone without us having to do any changes. I don't think it's possible with the current design of source generators, but if you have some idea, let me know.
- Extensibility for specific libraries which could be viable if CommunityToolkit.Mvvm publishes a new package that our generator will take a dependency on and then we call the library to get the info we need. This approach isn't any better than the current approach. It just moves the same code I wrote into a separate library and **adds a new dependency** which isn't good, and also still we haven't got the extensibility we would like.

I'm definitely open to any ideas you might have if you can think of an interesting approach that I couldn't think of, but for the most part, we can consider this PR as a workaround for https://github.com/dotnet/roslyn/issues/57239. If https://github.com/dotnet/roslyn/issues/57239 ever got fixed, we will just revert this.

Reply:
Let's discuss this again if the need arises for another library. In the meantime, let's at least abstract the new search behavior in a separate class and interface, so we can add more of similar lookups.